src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -9722,6 +9722,27 @@ public void Deconstruct(out int a, out string b)
                 Diagnostic(ErrorCode.ERR_NameNotInContext, "y2").WithArguments("y2").WithLocation(7, 18));
         }
 
+        [Fact]
+        public void MixedDeclarationAndAssignmentForbiddenZone()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        (int x1, (x1, int x2), x2) = (1, (2, 3), 4);
```

Comment:
>(int x1, (x1, int x2), x2) = (1, (2, 3), 4); [](http://example.com/codeflow?start=8&length=44)

I see nothing wrong in this code and it looks like it is working as I would expect it to work.